### ISSUE TITLE
Contents Text ware Deleted in Markdown Sources

### ISSUE THREAD

**Issue by peacock0803sz** [atsphinx/linebreak#1]
Thank you for creating this extension from my question/request!

But in a markdown source, contents body (text of the `<p>` tag) ware deleted 😕 
I added assertions to `test__it` (like bellow patch), then it failed only for markdown pattern 👀 

```patch
diff --git a/tests/test_it.py b/tests/test_it.py
index cab05aa..507552f 100644
--- a/tests/test_it.py
+++ b/tests/test_it.py
@@ -11,5 +11,7 @@ def test__it(app: SphinxTestApp):
     app.build()
     soup = BeautifulSoup((app.outdir / "index.html").read_text(), "lxml")
     assert soup.p.br
+    assert soup.p.text == "This isa pen."
     soup = BeautifulSoup((app.outdir / "index2.html").read_text(), "lxml")
     assert soup.p.br
+    assert soup.p.text == "This isa pen."
```

Fail log:
```
app = <SphinxTestApp buildername='html'>

    @pytest.mark.sphinx("html")
    def test__it(app: SphinxTestApp):
        """Test to pass."""
        app.build()
        soup = BeautifulSoup((app.outdir / "index.html").read_text(), "lxml")
        assert soup.p.br
        assert soup.p.text == "This isa pen."
        soup = BeautifulSoup((app.outdir / "index2.html").read_text(), "lxml")
        assert soup.p.br
>       assert soup.p.text == "This isa pen."
E       AssertionError: assert '' == 'This isa pen.'
E
E         - This isa pen.

tests/test_it.py:17: AssertionError
```

I'm not sure to Sphinx and MyST-Parser behaviors, but I found the node is detected as only newline character (`\n`)...

```patch
diff --git a/src/atsphinx/linebreak/__init__.py b/src/atsphinx/linebreak/__init__.py
index be91f1e..8deff2e 100644
--- a/src/atsphinx/linebreak/__init__.py
+++ b/src/atsphinx/linebreak/__init__.py
@@ -31,6 +31,7 @@ def inject_line_break(app: Sphinx, doctree: nodes.document):
             continue
         splitted = [(nodes.Text(t), line_break()) for t in text.split("\n")]
         items = [item for parts in splitted for item in parts]
+        print(f"{text=}\n{splitted=}\n{items=}")
         p = text.parent
         p.children = items[:-1]
 ```

stdout (in the markdown case):
```txt
text=<#text: '\n'>
splitted=[(<#text: ''>, <line_break: >), (<#text: ''>, <line_break: >)]
items=[<#text: ''>, <line_break: >, <#text: ''>, <line_break: >]
```

<details><summary>Full stdout</summary>

```
$ uv run pytest .
=========================================================================================== test session starts ============================================================================================
platform darwin -- Python 3.9.6, pytest-8.3.3, pluggy-1.5.0
rootdir: /Users/peacock/ghq/github.com/atsphinx/linebreak
configfile: pyproject.toml
plugins: anyio-4.6.2.post1
collected 1 item

tests/test_it.py F                                                                                                                                                                                   [100%]

================================================================================================= FAILURES =================================================================================================
_________________________________________________________________________________________________ test__it _________________________________________________________________________________________________

app = <SphinxTestApp buildername='html'>

    @pytest.mark.sphinx("html")
    def test__it(app: SphinxTestApp):
        """Test to pass."""
        app.build()
        soup = BeautifulSoup((app.outdir / "index.html").read_text(), "lxml")
        assert soup.p.br
        assert soup.p.text == "This isa pen."
        soup = BeautifulSoup((app.outdir / "index2.html").read_text(), "lxml")
        assert soup.p.br
>       assert soup.p.text == "This isa pen."
E       AssertionError: assert '' == 'This isa pen.'
E
E         - This isa pen.

tests/test_it.py:17: AssertionError
------------------------------------------------------------------------------------------- Captured stdout call -------------------------------------------------------------------------------------------
text=<#text: 'This is\na pen.'>
splitted=[(<#text: 'This is'>, <line_break: >), (<#text: 'a pen.'>, <line_break: >)]
items=[<#text: 'This is'>, <line_break: >, <#text: 'a pen.'>, <line_break: >]
text=<#text: '\n'>
splitted=[(<#text: ''>, <line_break: >), (<#text: ''>, <line_break: >)]
items=[<#text: ''>, <line_break: >, <#text: ''>, <line_break: >]
----------------------------------------------------------------------------------------- Captured stdout teardown -----------------------------------------------------------------------------------------
# testroot: root
# builder: html
# srcdir: /private/var/folders/hk/84vfrl6n5y9fltd1ck_6f_c00000gp/T/pytest-of-peacock/pytest-34/root
# outdir: /private/var/folders/hk/84vfrl6n5y9fltd1ck_6f_c00000gp/T/pytest-of-peacock/pytest-34/root/_build/html
# status:
Running Sphinx v7.4.7
loading translations [en]... done
myst v3.0.1: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions=set(), disable_syntax=[], all_links_external=False, links_external_new_tab=False, url_schemes=('http', 'https', 'mailto', 'ftp'), ref_domains=None, fence_as_directive=set(), number_code_blocks=[], title_to_header=False, heading_anchors=0, heading_slug_func=None, html_meta={}, footnote_transition=True, words_per_minute=200, substitutions={}, linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', enable_checkboxes=False, suppress_warnings=[], highlight_code_blocks=True)
building [mo]: targets for 0 po files that are out of date
writing output...
building [html]: targets for 2 source files that are out of date
updating environment: [new config] 2 added, 0 changed, 0 removed
reading sources... [ 50%] index
reading sources... [100%] index2

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
copying assets...
copying static files... done
copying extra files... done
copying assets: done
writing output... [ 50%] index
writing output... [100%] index2

generating indices... genindex done
writing additional pages... search done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 1 warning.

The HTML pages are in ../../../../../../private/var/folders/hk/84vfrl6n5y9fltd1ck_6f_c00000gp/T/pytest-of-peacock/pytest-34/root/_build/html.

# warning:
/private/var/folders/hk/84vfrl6n5y9fltd1ck_6f_c00000gp/T/pytest-of-peacock/pytest-34/root/index2.md: WARNING: document isn't included in any toctree

============================================================================================= warnings summary =============================================================================================
tests/test_it.py::test__it
  /Users/peacock/ghq/github.com/atsphinx/linebreak/.venv/lib/python3.9/site-packages/urllib3/__init__.py:35: NotOpenSSLWarning: urllib3 v2 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'LibreSSL 2.8.3'. See: https://github.com/urllib3/urllib3/issues/3020
    warnings.warn(

tests/test_it.py::test__it
tests/test_it.py::test__it
  /Users/peacock/ghq/github.com/atsphinx/linebreak/.venv/lib/python3.9/site-packages/bs4/builder/_lxml.py:124: DeprecationWarning: The 'strip_cdata' option of HTMLParser() has never done anything and will eventually be removed.
    parser = parser(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================= short test summary info ==========================================================================================
FAILED tests/test_it.py::test__it - AssertionError: assert '' == 'This isa pen.'
====================================================================================== 1 failed, 3 warnings in 0.41s =======================================================================================
```

</details>

**Comment by attakei** [comment]
## Reasons:

1. MyST splits sentence and linebreak when generating docutils' nodes.
2. I implemented to "replace" all children of parent(paragraph).

To see about (1), you can check by call `uv run pytest -s` after editing `src/atsphinx/linebreak/__init__.py`.

```patch
def inject_line_break(app: Sphinx, doctree: nodes.document):
+   print(doctree["source"])
+   print(next(doctree.findall(nodes.paragraph)).children)
    for text in doctree.findall(nodes.Text):
```

**Comment by attakei** [comment]
I reelased v0.1.2 to may improve this issue.
(it passes tests that is added your diff, and fixes other problem)

**Comment by peacock0803sz** [comment]
@attakei Thank you so much, I checked this issue was resolved with v0.1.2 🎉
